### ISSUE TITLE
[Ch 01][Notebook][Minor] FAQ Q6 sparse 메모리 예시가 실측과 어긋남 — nnz 50만·6MB·70배 (실측 405,803·4.9MB·82배)

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#88]
## 대상

- **챕터**: Ch 01 — 텍스트를 숫자로 (TF-IDF)
- **산출물**: Notebook (`01_tfidf/01_tfidf.ipynb` — `## ❓ FAQ` Q6 `cell023`)
- **기준 커밋**: `eed3dbe`
- **환경**: 정적 검토 + 실행본 `executed/01_tfidf.ipynb` 출력 대조 + 로컬 재실행(sklearn 1.9.0). **Colab 전체 실행 완료** (2026-08-15 · CPU 런타임 · 63초 · 에러 0건)

## 실제 결과

FAQ Q6 의 sparse 메모리 예시가 **이 챕터의 실측값과 어긋납니다.**

> sparse(CSR) 행렬은 0이 아닌 칸만 `(값, 열 인덱스)`로 저장합니다. **nnz가 50만이면 대략 `500000 × (8 + 4) ≈ 6MB`** — 70배 가까이 절약됩니다.

같은 챕터가 **두 셀 앞**(`cell013`)에서 실제 nnz 를 출력합니다.

| 항목 | 값 | 출처 |
|---|---|---|
| 본문 가정 nnz | 500,000 | FAQ Q6 |
| **실측 nnz** | **약 40만** (405,803 / 405,787) | `cell013` 출력 — 아래 참고 |
| 본문 계산 | 6MB · 약 70배 절약 | FAQ Q6 |
| **실측 계산** | **4.9MB · 약 82배 절약** | 아래 스니펫 |

FAQ 본문의 코드 스니펫을 그대로 돌리면 `4.9MB` / 약 82배가 나옵니다. 가정형("nnz가 50만이면")이라 명백한 오류는 아니지만, 학습자가 위 출력과 아래 스니펫 결과를 함께 보면 6MB·70배와 어긋나 보입니다.

### nnz 는 환경에 따라 미세하게 달라집니다 — 정확값을 본문에 박으면 안 됩니다

같은 5,000건·같은 코드인데 실행 환경에 따라 값이 갈렸습니다.

| 실행 | nnz |
|---|---|
| 실행본 `executed/01_tfidf.ipynb` (Colab, python 3.12.13) | 405,803 |
| 로컬 재실행 (sklearn 1.9.0) | 405,803 |
| **Colab 재실행 (2026-08-15)** | **405,787** |

**데이터는 동일합니다** — 같은 실행에서 `len_words.describe()` 의 `mean 133.811400` 이 실행본과 정확히 일치했습니다. 즉 `shuffle(seed=42)` 로 뽑힌 5,000건은 그대로이고, `max_features=10000` 의 어휘 컷 경계에 걸린 단어들이 sklearn 버전에 따라 갈리면서 nnz 만 ±수십 움직인 것으로 보입니다. `sparsity: 99.19%` 는 두 값 모두 동일합니다.

노트북이 `sklearn.__version__` 을 출력하지 않아 실행본이 어느 버전에서 나온 값인지 되짚을 수 없습니다.

## 기대 결과

위 셀의 출력과 이어지되, **버전에 따라 흔들리지 않는 표기**여야 합니다.

> "이 챕터의 nnz 는 약 40만이므로 대략 `400000 × (8 + 4) ≈ 4.9MB` — **80배 넘게 절약**됩니다."

정확값(405,803)을 박으면 학습자가 자기 출력(예: 405,787)과 다른 숫자를 보게 되어 지금 FAQ 와 같은 종류의 어긋남이 반복됩니다. #63 제안 1의 3단 규약에는 이 성격의 칸이 아직 없어, **"환경 의존"(seed 는 고정이나 라이브러리 버전에 흔들리는 값) 구분을 #63 에 제안**해 두었습니다 — `sparsity: 99.19%` 처럼 자릿수를 줄이면 안정적으로 맞습니다.

## 재현 절차

```python
# 노트북 실행 후 같은 커널에서
print(X_count.nnz)                                                            # 405803 또는 405787 (환경 의존)
print((X_count.data.nbytes + X_count.indices.nbytes + X_count.indptr.nbytes) / 1e6)  # 4.889
print(5000 * 10000 * 8 / 1e6)                                                 # 400.0
print(400.0 / 4.889)                                                          # 81.8
```

```bash
grep -n "50만\|6MB\|70배" 01_tfidf/01_tfidf.ipynb pages/01-tfidf-wrapup.md
```

## 로그

```
# 실행본 / 로컬 (sklearn 1.9.0)
non-zero entries: 405,803
dense  : 400.0 MB
sparse : 4.889 MB   (81.8배)

# Colab 재실행 (2026-08-15, CPU 런타임)
non-zero entries: 405,787
mean    133.811400          ← 실행본과 동일 (데이터는 같음)
```

## 전파 범위

| 산출물 | 상태 |
|---|---|
| `01_tfidf/01_tfidf.ipynb` `cell023` | ❌ |
| `pages/01-tfidf-wrapup.md` | ❌ |
| `book/chapters/ch01_tfidf.tex:526` | ❌ → EPUB 영향 |

## 심각도

**Minor** — 가정형 서술이라 그 자체로 틀린 문장은 아니고, 읽는 데 지장도 없습니다. 내용 동결(9/20) 전 수정 가능합니다.

## 재실행 여부

마크다운 셀 수정이므로 **노트북 재실행은 불필요**합니다. WikiDocs 재변환 → EPUB 재빌드가 필요합니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@modesta-jihye-lee)
- 연결: 챕터 검수 이슈 #25 · 표기 규약 논의 #63

**Comment by yoon-gu** [comment]
PASS — FAQ Q6 sparse 메모리 예시가 `약 40만` · `≈ 4.9MB` · `80배 넘게` 근사 표기로 교체됐습니다. 지적하신 대로 실행마다 흔들리는 nnz 정확값을 박지 않고 자릿수를 낮췄습니다.

— PR #289 (master `6986584`, 2026-09-18). `origin/master` 기준으로 노트북·`executed/`·`pages/`·`tex` 전파까지 확인했습니다.
